schemas/stsci.edu/asdf/0.1.0/time/time.yaml

```diff
@@ -0,0 +1,232 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "http://stsci.edu/schemas/asdf/0.1.0/time/time"
+title: Represents an instance in time.
+description: |
+  A "time" is a single instant in time.  It may explicitly specify the
+  way time is represented (the "format") and the "scale" which the
+  offset and scaling relation of the unit of time.
+
+  Specific emphasis is placed on supporting time scales (e.g. UTC,
+  TAI, UT1, TDB) and time representations (e.g. JD, MJD, ISO 8601)
+  that are used in astronomy and required to calculate, e.g., sidereal
+  times and barycentric corrections.
+
+  Times may be represented as one of the following:
+
+  - an object, with explicit `value`, and optional `format`, `scale`
+    and `location`.
+
+  - a string, in which case the format is guessed from across
+    the unambiguous options (`iso`, `byear`, `jyear`, `yday`), and the
+    scale is hardcoded to `UTC`.
+
+  In either case, a single time tag may be used to represent an
+  n-dimensional array of times, using either an `ndarray` tag or
+  inline as (possibly nested) YAML lists.  If YAML lists, the same
+  format must be used for all time values [TODO: This is a restriction
```

Comment:
I think this would severely impact performance.  Basically now you pass the entire array to each Format parser and it will give an exception as soon as something fails.  If you don't assume that the inputs are homogeneous then every array element needs to be passed through the possible format parsers in turn to guess the format.  So for an input which is in the last attempted format the code would basically need to fail N x (M-1) times for N elements and M formats.  Instead now you have more like M fails until getting to the N successful parses.

(Yes, I suppose you could be clever and re-arrange the guess order on the fly..., but I think even this would add a lot of overhead).